src/rpc/json.go

```diff
@@ -0,0 +1,12 @@
+package rpc
+
+type JSONRPC struct {
```

Comment:
Have you considered something like https://pkg.go.dev/net/rpc for implementing the rpc client? 

We'd have to code an `interface` and `types` for each method, but we'll get to skip out on handling the base layer of rpc parsing